rclpy/src/rclpy/node.cpp

```diff
@@ -441,7 +443,22 @@ Node::Node(
     new rcl_node_t,
     [](rcl_node_t * node)
     {
-      rcl_ret_t ret = rcl_node_fini(node);
+      rcl_ret_t ret;
+      {
+        rclpy::LoggingGuard scoped_logging_guard;
+        if (rcl_logging_rosout_enabled()) {
```

Comment:
we do not check the implementation, since `rcl_logging_rosout_fini_publisher_for_node` can return immediately if the impl is not constructed.